src/libraries/System.IO.Pipelines/tests/StreamPipeWriterTests.cs

```diff
@@ -592,6 +592,25 @@ public async Task OperationCancelledExceptionNotSwallowedIfNotThrownFromSpecifie
             await Assert.ThrowsAsync<OperationCanceledException>(async () => await writer.WriteAsync(new byte[1]));
         }
 
+        [Fact]
+        public void UnflushedBytesWorks()
+        {
+            byte[] bytes = Encoding.ASCII.GetBytes("Hello World");
+            var stream = new MemoryStream();
+            PipeWriter writer = PipeWriter.Create(stream);
+
+            Assert.True(writer.CanGetUnflushedBytes);
+
+            bytes.AsSpan().CopyTo(writer.GetSpan(bytes.Length));
+            writer.Advance(bytes.Length);
+
+            Assert.Equal(bytes.Length, writer.UnflushedBytes);
+
+            writer.Complete();
```

Comment:
I wonder if we should test that UnflushedBytes is zero after FlushAsync, WriteAsync and Complete for StreamPipeWriter and DefaultPipeWriter.

Reply:
yes we should

Reply:
I've added unflushed bytes checks on the tests where it looks like completion should work, i.e. not ones that throw.